src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCaptureCollection.cs

```diff
@@ -47,6 +50,25 @@ public int Count
         }
 
         /// <summary>
+        /// Copies all the elements of the collection to the given array
+        /// beginning at the given index.
+        /// </summary>
+        public void CopyTo(Capture[] array, int arrayIndex)
+        {
+            if (array == null)
+                throw new ArgumentNullException("array");
+            if (arrayIndex < 0 || arrayIndex > array.Length)
```

Comment:
Since we are already not very consistent, I would do either what you suggested (to simplify the code) or what I suggested (follow current convention and implemented correctly). I would not try to be consistent with incorrectly implemented range checks ;-)

Moreover, it would be great to have a helper for common range checks, so that people don't make similar mistakes (to those made in the existing collections). Something that can be called along the lines of:
Precondition.RequireFit(source : this, destination : array, destinationIndex : arrayIndex);